exporters/otlp/trace/src/main/java/io/opentelemetry/exporter/otlp/trace/OtlpHttpSpanExporterBuilder.java

```diff
@@ -0,0 +1,171 @@
+      throw new IllegalArgumentException(
+          "Invalid endpoint, must start with http:// or https://: " + uri);
+    }
+
+    this.endpoint = endpoint;
+    return this;
+  }
+
+  /** Sets the encoding of payloads. If unset, defaults to {@link #DEFAULT_ENCODING}. */
+  public OtlpHttpSpanExporterBuilder setEncoding(Encoding encoding) {
+    // TODO: if we we don't anticipate additional encodings, could replace this with
+    // setJsonEncoding()
+    requireNonNull(endpoint, "encoding");
+    this.encoding = encoding;
+    return this;
+  }
+
+  /**
+   * Sets the method used to compress payloads. If unset, compression is disabled. Currently the
+   * only supported compression method is "gzip".
+   */
+  public OtlpHttpSpanExporterBuilder setCompression(String compressionMethod) {
+    requireNonNull(endpoint, "compressionMethod");
+    Preconditions.checkArgument(
+        "gzip".equals(compressionMethod),
+        "Unsupported compression method. Supported compression methods include: gzip.");
+    this.isCompressionEnabled = true;
+    return this;
+  }
+
+  /**
+   * Sets the maximum time to wait for the collector to process an exported batch of spans. If
+   * unset, defaults to {@value DEFAULT_TIMEOUT_SECS}s.
+   */
+  public OtlpHttpSpanExporterBuilder setTimeout(long timeout, TimeUnit unit) {
+    requireNonNull(unit, "unit");
+    checkArgument(timeout >= 0, "timeout must be non-negative");
+    timeoutNanos = unit.toNanos(timeout);
+    return this;
+  }
+
+  /**
+   * Sets the maximum time to wait for the collector to process an exported batch of spans. If
+   * unset, defaults to {@value DEFAULT_TIMEOUT_SECS}s.
+   */
+  public OtlpHttpSpanExporterBuilder setTimeout(Duration timeout) {
+    requireNonNull(timeout, "timeout");
+    return setTimeout(timeout.toNanos(), TimeUnit.NANOSECONDS);
+  }
+
+  /** Add header to request. */
+  public OtlpHttpSpanExporterBuilder addHeader(String key, String value) {
+    if (headers == null) {
+      headers = new HashMap<>();
+    }
+    headers.computeIfAbsent(key, (k) -> new ArrayList<>()).add(value);
+    return this;
+  }
+
+  /**
+   * Sets the certificate chain to use for verifying servers when TLS is enabled. The {@code byte[]}
+   * should contain an X.509 certificate collection in PEM format. If not set, TLS connections will
+   * use the system default trusted certificates.
+   */
+  public OtlpHttpSpanExporterBuilder setTrustedCertificates(byte[] trustedCertificatesPem) {
+    this.trustedCertificatesPem = trustedCertificatesPem;
+    return this;
+  }
+
+  /**
+   * Constructs a new instance of the exporter based on the builder's values.
+   *
+   * @return a new exporter's instance
+   */
+  public OtlpHttpSpanExporter build() {
+    OkHttpClient.Builder clientBuilder =
+        new OkHttpClient.Builder()
+            // TODO: which thread should handle these calls?
+            .dispatcher(new Dispatcher(MoreExecutors.newDirectExecutorService()))
```

Comment:
Noticed that the `OtlpGrpcSpanExporter` executes using `MoreExecutors.directExecutor()` and followed that pattern here. Other options would be to execute requests in a blocking fashion or execute them on another group of threads.

Reply:
grpc executes user callbacks on the direct executor, not the requests themselves which are always async on netty event loop.

IIRC this okhttp knob actually controls what threads requests run on, which we definitely don't want on the hot path. So let's leave this at the default.